### ISSUE TITLE
surveillance:type=* should be present if camera:*=* is

### ISSUE THREAD

**Issue by ivanbranco** [osm-fr/osmose-backend#1840]
Hi,

I would like to propose a new check. If an element is tagged with a camera:*=* prefix, it should have a surveillance:type=* value as well, usually =camera or =ALPR

Overpass query: https://overpass-turbo.eu/s/1upC

**Comment by Famlam** [comment]
https://wiki.openstreetmap.org/wiki/Key:camera:type
Note it can also be used on `shop=camera`.

I personally would not be too enthusiastic about this test. I'm just thinking along the lines of: 
- if I were to see a camera on the street, would I be able to see if it's scanning number plates or capturing general video footage? -> probably not, they look fairly similar.
- However, I am very well able to see if a camera is a dome or not (camera:type), whether it's pole mounted or not (camera:mount). But if I do add that, I would get extra warnings for missing `surveillance:type`.

**Comment by ivanbranco** [comment]
> [https://wiki.openstreetmap.org/wiki/Key📷type](https://wiki.openstreetmap.org/wiki/Key:camera:type)
> Note it can also be used on shop=camera.

This could be avoided by exclusing camera:type and using only camera:mount, camera:angle and camera:direction

> I would get extra warnings for missing surveillance:type.

Should this be a condition to keep in mind for warnings? I'm full of warnings that I will never fix just because I corrected a typo in the name somewhere where I can't check stuff on the ground. With this logic I shouldn't fix typos because I would get a warning.

**Comment by Famlam** [comment]
Agree, that last sentence should've been phrased differently, I'll remove it as it makes a wrong point.

**Comment by Marc-marc-marc** [comment]
I don't see why surveillance:type=* is mandatory (otherwise with this logic, building:levels should be mandatory with building, and everyone should be able to add the info they want in mandatory from another tag)
or a level 4 "mapcomplete"should be created (and disabled by default) to list all the current missing tag suggestions, a bit like StreetComplete

**Comment by ivanbranco** [comment]
> otherwise with this logic, building:levels should be mandatory with building

This looks kinda the opposite to me, is like having building:levels=* without building=*
If you have a tag that describes the building levels, but you don't have the building, something is wrong, don't you agree?

Same here, you have a tag that describes the camera properties, but you don't have the camera, since man_made=surveillance alone could be also a guard, to say one.

**Comment by Famlam** [comment]
If it's such a must-be-together combination we would probably forward you to JOSM to get it in their (and, after a synchronization, our) mapcss rules ;) .

I do stand by my point that it's not a good idea to add a rule for this as it's an (often) unverifiable property.
`camera:mount`/`camera:type` etc implies it's a camera of any type. Whether it's a licence plate scanner camera or a general camera, as defined in `surveillance:type`, is, as Marc says, a nice addition, but not a must IMO.
